--- a/support/download/cargo-post-process	2025-01-09 14:43:35.000000000 +0000
+++ b/support/download/cargo-post-process	2025-01-31 14:50:13.755650009 +0000
@@ -38,7 +38,7 @@
 cargo vendor \
     --manifest-path "${manifest}" \
     --locked VENDOR \
-    > .cargo/config
+    >> .cargo/config
 
 # "cargo vendor' outputs on stderr a message directing to add some data
 # to the project's .cargo/config.toml, data that it outputs on stdout.
