From 11b63ef4e71e30e3736cd69a00b8d0618a7867b7 Mon Sep 17 00:00:00 2001
From: Sam Edwards <CFSworks@gmail.com>
Date: Sat, 20 May 2023 20:53:14 -0600
Subject: [PATCH 2/2] gpio: aggregator: Add "turingmachines,tp2-nodes"
 compatible

---
 drivers/gpio/gpio-aggregator.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpio/gpio-aggregator.c b/drivers/gpio/gpio-aggregator.c
index 38e0fff9afe7..e128be5d13eb 100644
--- a/drivers/gpio/gpio-aggregator.c
+++ b/drivers/gpio/gpio-aggregator.c
@@ -605,6 +605,7 @@ static const struct of_device_id gpio_aggregator_dt_ids[] = {
 	 * Add GPIO-operated devices controlled from userspace below,
 	 * or use "driver_override" in sysfs.
 	 */
+    { .compatible = "turing,pi2-nodes", },
 	{}
 };
 MODULE_DEVICE_TABLE(of, gpio_aggregator_dt_ids);
-- 
2.45.2

