From 81d9505285d3ac75b26d7db97ff88b3bb4c3280c Mon Sep 17 00:00:00 2001
From: Sam Edwards <CFSworks@gmail.com>
Date: Sat, 20 May 2023 20:53:14 -0600
Subject: [PATCH 3/3] gpio: aggregator: Add "turingmachines,tp2-nodes"
 compatible

---
 drivers/gpio/gpio-aggregator.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpio/gpio-aggregator.c b/drivers/gpio/gpio-aggregator.c
index 9b0adbdddbfc..92afa9ce8a77 100644
--- a/drivers/gpio/gpio-aggregator.c
+++ b/drivers/gpio/gpio-aggregator.c
@@ -536,6 +536,7 @@ static const struct of_device_id gpio_aggregator_dt_ids[] = {
 	 * Add GPIO-operated devices controlled from userspace below,
 	 * or use "driver_override" in sysfs
 	 */
+	{ .compatible = "turing,pi2-nodes", },
 	{}
 };
 MODULE_DEVICE_TABLE(of, gpio_aggregator_dt_ids);
-- 
2.39.2

