From 591ceb86fba53d44d0f88de9bd4ba79e810a6c97 Mon Sep 17 00:00:00 2001
From: Sam Edwards <CFSworks@gmail.com>
Date: Mon, 12 Jun 2023 13:59:38 -0600
Subject: [PATCH 16/17] HACK: basic spl->fel boot support

---
 arch/arm/mach-sunxi/board.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm/mach-sunxi/board.c b/arch/arm/mach-sunxi/board.c
index d493075fc0..829d170aeb 100644
--- a/arch/arm/mach-sunxi/board.c
+++ b/arch/arm/mach-sunxi/board.c
@@ -210,6 +210,16 @@ static int spl_board_load_image(struct spl_image_info *spl_image,
 	return 0;
 }
 SPL_LOAD_IMAGE_METHOD("FEL", 0, BOOT_DEVICE_BOARD, spl_board_load_image);
+
+static void go_to_fel(void)
+{
+	/* change lr to the well-known fel entry point */
+	fel_stash.lr &= ~0xFFFF;
+	fel_stash.lr |= 0x0020;
+
+	debug("Entering FEL sp=%x, lr=%x\n", fel_stash.sp, fel_stash.lr);
+	return_to_fel(fel_stash.sp, fel_stash.lr);
+}
 #endif /* CONFIG_SPL_BUILD */
 
 #define SUNXI_INVALID_BOOT_SOURCE	-1
@@ -454,8 +464,16 @@ u32 spl_mmc_boot_mode(struct mmc *mmc, const u32 boot_device)
 	return result;
 }
 
+#define SUNXI_RTC_BASE 0x07090000
 void board_init_f(ulong dummy)
 {
+	uintptr_t fel_flag = SUNXI_RTC_BASE + 0x100 + 2*4;
+	if (readl(fel_flag) == 0x5AA5A55A) {
+		writel(0, fel_flag);
+		go_to_fel();
+		return;
+	}
+
 	sunxi_sram_init();
 
 #if defined CONFIG_MACH_SUN6I || defined CONFIG_MACH_SUN8I_H3
-- 
2.44.0

